File: 1030A -In Search of an Easy Problem/InSearchOfAnEasyProblem.c
#include <stdio.h>
#include <stdbool.h>

int main() {
    int n, opinion;
    bool isEasy = true;

    scanf("%d", &n);

    for(int i = 0; i < n; i++) {
        scanf("%d", &opinion);
        if (opinion == 1) {
            isEasy = false;
        }
    }

    if (isEasy) {
        printf("EASY\n");
    } else {
        printf("HARD\n");
    }
    return 0;
}

File: 1030A -In Search of an Easy Problem/InSearchOfAnEasyProblem.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

func main() {
	var scanner *bufio.Scanner
	var lines []string
	var isEasy bool = true

	scanner = bufio.NewScanner(os.Stdin)

	for scanner.Scan() {
		lines = append(lines, scanner.Text())
	}

	for _, str := range strings.Fields(lines[1]) {
		if str == "1" {
			isEasy = false
		}
	}

	if isEasy {
		fmt.Println("EASY")
	} else {
		fmt.Println("HARD")
	}
}
